### ISSUE TITLE
你好，我这边利用代码进行压缩与重建，不过会随机出现重建出现错误的情况，关键这个随机性我实在摸不到头脑，具体的代码和重建失败的图像我提供在下面

### ISSUE THREAD

**Issue by SpiritPiao** [mmSir/GainedVAE#7]

**Comment by SpiritPiao** [comment]
def _encode(model, x, s, l):
    model.eval()
    x = x.unsqueeze(0)
    p =64
    x_padded, padding = pad(x, p)

    start = time.time()
    out_enc = model.compress(x_padded, s=s, l=l)
    out_forward = model(x_padded, s=s) if l != 1.0 else model(x_padded, s=s + 1)
    enc_time = time.time() - start

    start = time.time()

    out_dec = model.decompress(out_enc['strings'], out_enc['shape'], s=s, l=l)
    y_quantized = out_dec["gained_y_hat"]  # gained_y_quantized
    dec_time = time.time() - start

    out_dec["x_hat"] = crop(out_dec["x_hat"], padding)

    num_pixels = x.size(0) * x.size(2) * x.size(3)
    bpp = sum(len(s[0]) for s in out_enc["strings"]) * 8.0 / num_pixels
    estimate_bpp = sum((torch.log(likelihoods).sum() / (-math.log(2) * num_pixels))
                       for likelihoods in out_forward["likelihoods"].values()).item()

    return {
        "x_hat": out_dec["x_hat"],
        "string":out_enc["strings"],
        "shape": out_enc["shape"],
        "padding":padding,
        "y_quantized": y_quantized,
        "psnr": psnr(x, out_dec["x_hat"]),
        "ms-ssim": ms_ssim(x, out_dec["x_hat"], data_range=1.0).item(),
        "bpp": bpp,
        "estimate_bpp": estimate_bpp,
        "encoding_time": enc_time,
        "decoding_time": dec_time,
    }
